custom_components/google_home/entity.py

```diff
@@ -1,208 +1,66 @@
+        return f"{self.device_name} {self.label}"
 
     @property
     def unique_id(self) -> str:
         """Return a unique ID to use for this entity."""
-        return f"{self.device_name}/{LABEL_NEXT_ALARM}"
-
-    @property
-    def icon(self) -> str:
-        """Icon to use in the frontend, if any."""
-        return ICON_ALARMS
-
-    @property
-    def device_class(self) -> str:
-        """Return the device class of the sensor."""
-        return DEVICE_CLASS_TIMESTAMP
-
-
-class GoogleHomeTimersEntity(CoordinatorEntity):
-    """Entity base for timers sensor"""
-
-    def __init__(self, coordinator: DataUpdateCoordinator, device_name: str):
-        super().__init__(coordinator)
-        self.device_name = device_name
-
-    @property
-    def name(self) -> str:
-        """Return the name of the sensor."""
-        return f"{self.device_name} {LABEL_TIMERS}"
+        return f"{self.device_name}/{self.label}"
 
     @property
     def device_info(self) -> DeviceInfo:
         return {
             "identifiers": {(DOMAIN, self.device_name)},
             "name": f"{DEFAULT_NAME} {self.device_name}",
-            "model": VERSION,
             "manufacturer": MANUFACTURER,
         }
 
-    @property
-    def unique_id(self) -> str:
-        """Return a unique ID to use for this entity."""
-        return f"{self.device_name}/{LABEL_TIMERS}"
-
-    @property
-    def icon(self) -> str:
-        """Icon to use in the frontend, if any."""
-        return ICON_TIMERS
-
-
-class GoogleHomeNextTimerEntity(CoordinatorEntity):
-    """Entity base for next timer sensor"""
-
-    def __init__(self, coordinator: DataUpdateCoordinator, device_name: str):
-        super().__init__(coordinator)
-        self.device_name = device_name
-
-    @property
-    def device_info(self) -> DeviceInfo:
-        return {
-            "identifiers": {(DOMAIN, self.device_name)},
-            "name": f"{DEFAULT_NAME} {self.device_name}",
-            "model": VERSION,
-            "manufacturer": MANUFACTURER,
-        }
-
-    @property
-    def name(self) -> str:
-        """Return the name of the sensor."""
-        return f"{self.device_name} {LABEL_NEXT_TIMER}"
-
-    @property
-    def unique_id(self) -> str:
-        """Return a unique ID to use for this entity."""
-        return f"{self.device_name}/{LABEL_NEXT_TIMER}"
-
-    @property
-    def icon(self) -> str:
-        """Icon to use in the frontend, if any."""
-        return ICON_TIMERS
-
-    @property
-    def device_class(self) -> str:
-        """Return the device class of the sensor."""
-        return DEVICE_CLASS_TIMESTAMP
+    def get_device(self):
+        """Return the device matched by device name
+        from the list of google devices in coordinator_data"""
+        matched_devices = [
+            device
+            for device in self.coordinator.data
+            if device.name == self.device_name
+        ]
+        return matched_devices[0] if matched_devices else None
```

Comment:
Love it! Thanks for implementing it. Maybe not the most looking solution, but hopefully more robust